src/engine/source/indexerconnector/src/indexerConnector.cpp

```diff
@@ -9,23 +9,92 @@
  * Foundation.
  */
 
-#include "indexerConnector/indexerConnector.hpp"
-#include "HTTPRequest.hpp"
-#include "base/logging.hpp"
-#include "base/utils/stringUtils.hpp"
-#include "base/utils/timeUtils.hpp"
+#include <filesystem>
+#include <fstream>
+#include <grp.h>
+#include <pwd.h>
+#include <unistd.h>
+
+#include <HTTPRequest.hpp>
+#include <base/logging.hpp>
+#include <base/utils/stringUtils.hpp>
+#include <base/utils/timeUtils.hpp>
+#include <indexerConnector/indexerConnector.hpp>
+
 #include "secureCommunication.hpp"
 #include "serverSelector.hpp"
-#include <fstream>
 
 constexpr auto INDEXER_COLUMN {"indexer"};
 constexpr auto USER_KEY {"username"};
 constexpr auto PASSWORD_KEY {"password"};
 constexpr auto ELEMENTS_PER_BULK {1000};
+constexpr auto WAZUH_OWNER {"wazuh"};
+constexpr auto DEFAULT_PATH {"tmp/root-ca-merged.pem"};
 
 // Single thread in case the events needs to be processed in order.
 constexpr auto SINGLE_ORDERED_DISPATCHING = 1;
 
+/**
+ * @brief Merges the CA root certificates into a single file.
+ * @param filePaths The list of CA root certificates file paths.
+ * @param caRootCertificate The path to the merged CA root certificate.
+ * @throws std::runtime_error If the CA root certificate file does not exist, could not be opened, written or the
+ * ownership could not be changed.
+ * @note The merged file will be created in the same directory as the first file in the list.
+ */
+static void mergeCaRootCertificates(const std::vector<std::string>& filePaths, std::string& caRootCertificate)
+{
+    std::string caRootCertificateContentMerged;
+
+    for (const auto& filePath : filePaths)
+    {
+        if (!std::filesystem::exists(filePath))
+        {
+            throw std::runtime_error("The CA root certificate file: '" + filePath + "' does not exist.");
+        }
+
+        std::ifstream file(filePath);
+        if (!file.is_open())
+        {
+            throw std::runtime_error("Could not open CA root certificate file: '" + filePath + "'.");
+        }
+
+        caRootCertificateContentMerged.append((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+    }
+
+    caRootCertificate = DEFAULT_PATH;
```

Comment:
This cannot be configured so they are the only path

Reply:
Done, thanks!